pkg/config/config.go

```diff
@@ -151,6 +151,7 @@ type GasEstimator interface {
 	Mode() string
 	PriceMaxKey(gethcommon.Address) *assets.Wei
 	EstimateLimit() bool
+	SenderAddress() *string
```

Comment:
Let's return an address instead of a string. Then we get validation once up-front during initialization and the callers don't have to worry about it.
```suggestion
	SenderAddress() *gethcommon.Address
```

Reply:
That makes sense, but if we were to return an address instead of a string, then there would be no way to leave the parameter un-set if we don't want to set a custom estimation sender address (like we are doing now with an empty string value)

Reply:
We use pointers for every field so that we can tell if it was not set.

Reply:
Thank you for your guidance here Jordan, I've updated the type of the parameter in the PR